app/services/true_layer/bank_data_import_service.rb

```diff
@@ -0,0 +1,45 @@
+module TrueLayer
+  class BankDataImportService
+    def initialize(applicant:, token:)
+      @applicant = applicant
+      @token = token
+    end
+
+    def call
+      return unless bank_provider
+
+      import_bank_accounts
+      import_account_holders
```

Comment:
⚠️ There's a chance that if any of the these steps fail, while a previous step succeeded, given they're not wrapped in a transaction, that we might end up with inconsistent data (e.g. imported bank account and holders but the transactions pull fails).

💭 Consider having some process in place to trigger a rollback so that we either import everything or rollback to the initial state, to guarantee consistency of data.

Reply:
I'm thinking of refactoring the code to do all API requests first and then do the DB changes. That way we don't even need transactions and most importantly, we avoid long deadlocks caused by API requests in-between DB calls.
What do you think?